backend/src/main/java/io/dcbn/backend/passwordReset/models/MailType.java

```diff
@@ -1,5 +1,29 @@
 package io.dcbn.backend.passwordReset.models;
 
+import io.dcbn.backend.authentication.models.DcbnUser;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.security.Keys;
+import java.util.Calendar;
+import org.springframework.stereotype.Component;
+
+@Component
 public enum MailType {
-  PASSWORD_RESET
+  PASSWORD_RESET {
+    @Override
+    public String generateMailBody(DcbnUser user, String secret) {
+      Calendar now = Calendar.getInstance();
+      now.add(Calendar.MINUTE, TOKEN_DURATION_MINUTES);
+
+      return Jwts.builder()
+          .signWith(Keys.hmacShaKeyFor(secret.getBytes()), SignatureAlgorithm.HS512)
+          .setSubject("" + user.getId())
+          .setExpiration(now.getTime())
+          .compact();
+    }
+  };
+
+  private static final int TOKEN_DURATION_MINUTES = 30;
+
+  public abstract String generateMailBody(DcbnUser user, String secret);
```

Comment:
Kann man das Secret auch iwi direkt in die klasse machen. Ist komisch das als parameter zu übergeben, oder? Also weil das ja nicht für jeden MailType benötigt wird

Reply:
Das Problem ist, dass das die Enums statisch sind und Spring da Probleme beim Autowiring hat. Man könnte eine andere Klasse erstellen die von Spring initialisiert wird und dann den Wert setzt. Siehe https://stackoverflow.com/a/21119815